### PR TITLE
drivers/video: Add proper error handling in failures

### PR DESCRIPTION
**Pull request by sijupunnnoose**
- Error handling for change video state, set buffer and type
and transfer failure cases.
- Merged function set_buf and set_buftype as single function.

Signed-off-by: Siju Cherian <siju.cherian@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [79ca6b639d44b11173f571c92d97c60e1aada637] - Code Rule Check Result:
os/drivers/video/video.c:320: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:324: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:532: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:539: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:543: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:590: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:614: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:892: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:900: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:904: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [79ca6b639d44b11173f571c92d97c60e1aada637] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cf23f1c96d9b5d6f0565aaab8bf02227f34f008c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cf23f1c96d9b5d6f0565aaab8bf02227f34f008c] - Code Rule Check (C++) OK.

File: os/drivers/video/video.c [hunk 1]
```diff
@@ -518,13 +527,22 @@ static int video_qbuf(FAR video_upperhalf_t *priv, FAR struct v4l2_buffer *buf)
...
 		video_unlock(&priv->still_inf.lock_state);
+		if (ret < 0) {
+			return ret;
+		}
 	} else {
 		container = video_framebuff_get_dma_container(&type_inf->bufinf);
 		if (container) {
-			video_devops->set_buftype(priv->dev, buf->type);
-			video_devops->set_buf(priv->dev, container->buf.m.userptr, container->buf.length);
+			ret = video_devops->set_buftype(priv->dev, buf->type);
```
Comment (Taejun-Kwon) [inline os/drivers/video/video.c]: video_unlock is unncessary? (it locked in line 519)

Comment (sijupunnnoose) [reply os/drivers/video/video.c]: Video lock is taken when changing states, and same is unlocked at line number 522.

Comment (Taejun-Kwon) [reply os/drivers/video/video.c]: if state is VIDEO_STATE_STREAMON, type_inf->lock_state is still locked here, and it returns ret.

**Comment by seinfra** [comment]
Target : [bc83afc306945a5dc4f656f4d0b781bad8ba06e1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bc83afc306945a5dc4f656f4d0b781bad8ba06e1] - Code Rule Check (C++) OK.

File: os/drivers/video/video.c (hunk 1 above)
Comment (sijupunnnoose) [reply os/drivers/video/video.c]: Corrected!

**Comment by seinfra** [comment]
Target : [b4d3a80f604fb2a4e443fcaab47df32536c25a3f] - Code Rule Check Result:
os/drivers/video/video.c:320: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:891: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video_null_driver.c:53: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:53: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:53: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:54: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:54: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:54: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/drivers/video/video_null_driver.c:55: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:55: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:56: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:56: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:57: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:57: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:58: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:58: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:59: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:59: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:60: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:60: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:61: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:62: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:62: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:63: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:63: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:64: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:66: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:67: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:67: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:68: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:68: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:69: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:173: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:174: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:175: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:176: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:177: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:178: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:179: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:184: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:185: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:186: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:187: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:188: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:189: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:190: ERROR: [SPC_M_SEP] space required after that close brace '}'

**Comment by seinfra** [comment]
Target : [b4d3a80f604fb2a4e443fcaab47df32536c25a3f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9e703af60243498782d9c45847fbb595648fa6e2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9e703af60243498782d9c45847fbb595648fa6e2] - Code Rule Check Result:
os/drivers/video/video.c:320: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video.c:891: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/video/video_null_driver.c:53: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:53: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:53: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:54: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:54: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:54: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/drivers/video/video_null_driver.c:55: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:55: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:56: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:56: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:57: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:57: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:58: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:58: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:59: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:59: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:60: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/video/video_null_driver.c:60: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:61: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:62: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:62: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:63: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:63: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:64: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:66: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/drivers/video/video_null_driver.c:67: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:67: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:68: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:68: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:69: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/video/video_null_driver.c:173: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:174: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:175: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:176: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:177: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:178: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:179: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:184: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:185: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:186: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:187: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:188: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:189: ERROR: [SPC_M_SEP] space required after that close brace '}'
os/drivers/video/video_null_driver.c:190: ERROR: [SPC_M_SEP] space required after that close brace '}'

**Comment by seinfra** [comment]
Target : [1956af0dbd3e553a7ea655e9ac2659e0cc9a5de3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1956af0dbd3e553a7ea655e9ac2659e0cc9a5de3] - Code Rule Check Result:
os/drivers/video/video_null_driver.c:54: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/drivers/video/video_null_driver.c:62: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:63: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:67: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('
os/drivers/video/video_null_driver.c:68: ERROR: [SPC_M_SEP] space prohibited between function name and open parenthesis '('

**Comment by seinfra** [comment]
Target : [9072467aada9a2fa871e35ca0d69fc4320782ccb] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9072467aada9a2fa871e35ca0d69fc4320782ccb] - Code Rule Check OK.

File: os/drivers/video/video.c [hunk 2]
```diff
@@ -315,8 +316,10 @@ static void change_video_state(FAR video_upperhalf_t *priv, enum video_state_e n
...
 		if (dma_container) {
-			video_devops->set_buftype(priv->dev, V4L2_BUF_TYPE_VIDEO_CAPTURE);
-			video_devops->set_buf(priv->dev, dma_container->buf.m.userptr, dma_container->buf.length);
+			ret = video_devops->set_buf(priv->dev, V4L2_BUF_TYPE_VIDEO_CAPTURE, dma_container->buf.m.userptr, dma_container->buf.length);
+			if (ret < OK) {
+				return ret;
+			}
 		} else {
 			updated_next_state = VIDEO_STATE_STREAMON;
 		}
```
Comment (jeongarmy) [inline os/drivers/video/video.c]: how about changing like below?
line 319
ret = video_devops->...
 [remove 320~322 line]

line 332
if (ret == OK) {
  priv->video_inf.state = updated_next_state;
}
return ret;

File: os/drivers/video/video_null_driver.c [hunk 3]
```diff
@@ -172,24 +169,24 @@ struct video_devops_s dummy_null_video_ops = {
...
-			{VIDEO_HSIZE_QVGA, VIDEO_VSIZE_QVGA, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,},
-			{VIDEO_HSIZE_VGA, VIDEO_VSIZE_VGA, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,},
-			{VIDEO_HSIZE_QUADVGA, VIDEO_VSIZE_QUADVGA, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,},
-			{VIDEO_HSIZE_HD, VIDEO_VSIZE_HD, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,},
-			{VIDEO_HSIZE_FULLHD, VIDEO_VSIZE_FULLHD, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,},
-			{VIDEO_HSIZE_5M, VIDEO_VSIZE_5M, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,},
-			{VIDEO_HSIZE_3M, VIDEO_VSIZE_3M, V4L2_FRMIVAL_TYPE_DISCRETE, {{333333,}, {500000,}, {666666,}, {1000000,}, {999999,}}, MAX_INTERVAL,}
+			{VIDEO_HSIZE_QVGA, VIDEO_VSIZE_QVGA, V4L2_FRMIVAL_TYPE_DISCRETE, { {333333,}, {500000,}, {666666,}, {1000000,}, {999999,} }, MAX_INTERVAL,},
```
Comment (jeongarmy) [inline os/drivers/video/video_null_driver.c]: why do you add space?

**Comment by seinfra** [comment]
Target : [9d63a60317d24fd676784e6c9a628e965a5f1d91] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9d63a60317d24fd676784e6c9a628e965a5f1d91] - Code Rule Check OK.

File: os/drivers/video/video.c (hunk 2 above)
Comment (sijupunnnoose) [reply os/drivers/video/video.c]: Corrected in set_buftype & set_buf unify patch.

File: os/drivers/video/video_null_driver.c (hunk 3 above)
Comment (sijupunnnoose) [reply os/drivers/video/video_null_driver.c]: os/drivers/video/video_null_driver.c:173: ERROR: [SPC_M_SEP] space required after that close brace '}'
I just adjusted the space in open  } also for better readability. Should we remove it or not?

File: os/drivers/video/Kconfig [hunk 4]
```diff
@@ -25,3 +25,8 @@ config FB_OVERLAY_BLIT
 config VIDEO_SOURCE
 	bool "Driver for Video Source"
 	default n
+
+config VIDEO_NULL
+	bool "Driver for Dummy Video lowerhalf"
+	depends on VIDEO_SOURCE
+	default n
```
Comment (sangam-swami) [inline os/drivers/video/Kconfig]: can we add new line here to remove this warning

**Pull request merged**